src/wallet/wallet.cpp

```diff
@@ -2659,6 +2659,12 @@ bool CWallet::CreateTransaction(interfaces::Chain::Lock& locked_chain, const std
 
             // Get the fee rate to use effective values in coin selection
             CFeeRate nFeeRateNeeded = GetMinimumFeeRate(*this, coin_control, &feeCalc);
+            // Do not, ever, assume that it's fine to change the fee rate if the user has explicitly
+            // provided one
+            if (coin_control.m_feerate && nFeeRateNeeded > *coin_control.m_feerate) {
+                strFailReason = strprintf("Fee rate is too low for the current mempool. Increase it to %s or modify -maxmempool", nFeeRateNeeded.ToString());
```

Comment:
The mempool fee is not considered when the user picks the fee rate. See also https://doxygen.bitcoincore.org/wallet_2fees_8cpp_source.html#l00028

How did you trigger this bug? It would be nice to include a test vector or steps to reproduce, otherwise this will just be dead code.

Reply:
@MarcoFalke the error is triggered by setting a fee rate below 1 sat / byte in the updated `sendtoaddress` RPCs in #11413. See https://github.com/bitcoin/bitcoin/pull/11413#issuecomment-593284781

Reply:
If it is not possible to hit this case with Bitcoin Core master, it probably makes sense to only add it when it can be hit. And also add a test to hit it.

Reply:
I triggered it with no problem in bitcoin-qt (set fee to 0.00000001 btc/kb)...

<img width="744" alt="Screen Shot 2020-03-07 at 10 20 51" src="https://user-images.githubusercontent.com/250224/76133784-383d7c00-605d-11ea-847b-c6f1596ceb51.png">

Without this patch, it goes to confirm screen, silently changing my fee rate:
<img width="751" alt="Screen Shot 2020-03-07 at 10 22 48" src="https://user-images.githubusercontent.com/250224/76133814-6c18a180-605d-11ea-8876-14558f79240e.png">

Reply:
> The mempool fee is not considered when the user picks the fee rate. See also https://doxygen.bitcoincore.org/wallet_2fees_8cpp_source.html#l00028
> 
> How did you trigger this bug? It would be nice to include a test vector or steps to reproduce, otherwise this will just be dead code.

That code calls `mempoolMinFee()` though.

Reply:
> That code calls mempoolMinFee() though.

No? It doesn't.

The code is

```cpp
         // Obey mempool min fee when using smart fee estimation
         CFeeRate min_mempool_feerate = wallet.chain().mempoolMinFee();
```

Setting a fee manually in the gui is not "smart fee". What am I missing?

Reply:
You're not missing anything, because you're absolutely right. Should've read the code more carefully...

I've fixed the message.